src/consumption.cpp

```diff
@@ -1500,28 +1500,42 @@ bool Character::fuel_bionic_with( item &it )
 
     const bionic_id bio = get_most_efficient_bionic( get_bionic_fueled_with( it ) );
 
-    const int loadable = std::min( it.charges, get_fuel_capacity( it.typeId() ) );
-    const std::string str_loaded  = get_value( it.typeId().str() );
+    const bool is_magazine = !!it.type->magazine;
```

Comment:
Why the double `!` ?

Reply:
It's the easiest way to get the `bool` I wanted, and follows what is done in `item::is_magazine()` and its friends.

Is there a preferred alternative?

Reply:
There's precedent; it's used elsewhere in the codebase so it's probably fine.  But the alternative would be `it.type->magazine != nullptr`, which might be clearer.  It's a question of personal taste.

Reply:
To me it looked like a typo ^^" so I feel like `it.type->magazine != nullptr` would be clearer.